docs/guides/using_link_props.rst

```diff
@@ -0,0 +1,123 @@
+.. _ref_guide_linkprops:
+
+=====================
+Using link properties
+=====================
+
+:index: property
+
+
+Links can contain **properties**. Due to how they're persisted under the hood,
+link properties have a few additional constraints: they're always *single* and
+*optional*.
+
+Declaration
+-----------
+
+.. code-block:: sdl
+
+  type Person {
+    required property name -> str { constraint exclusive; }
+
+    multi link friends -> Person {
+      property strength -> float64;
+    }
+  }
+
+Constraints
+-----------
+
+.. code-block:: sdl
+
+  type Person {
+    required property name -> str { constraint exclusive; }
+
+    multi link friends -> Person {
+      property strength -> float64;
+      constraint expression on (
+        __subject__@strength >= 0
+      );
+    }
+  }
+
+Indexes
+-------
+
+To index on a link property, you must declare an abstract link and extend it.
+
+.. code-block:: sdl
+
+  abstract link friendship {
+    property strength -> float64;
+    index on (__subject__@strength);
+  }
+
+  type Person {
+    required property name -> str { constraint exclusive; }
+    multi link friends extending friendship -> Person;
+  }
+
+
+Inserting
+---------
+
+.. code-block:: edgeql
+
+  insert Person {
+    name := "Bob",
+    friends := (
+      select detached Person {
+        @strength := 3.14
+      }
+      filter .name = "Alice"
+    )
+  }
```

Comment:
Perhaps we need to link `detached` docs here or have a sentence explaining why? Thoughts?

Reply:
👍👍👍